rebasehelper/output_tools/text_output_tool.py

```diff
@@ -125,35 +151,29 @@ def print_summary(cls, path, results):
         cls.results_store = results
 
         cls.print_success_message()
-        type_pkgs = ['old', 'new']
+        logger_report.info("Rebase helper results are located in %s", os.path.dirname(path))
+
+        cls.print_checkers_text_output(results.get_checkers())
+
         if results.get_patches():
             cls.print_changes_patch()
-            cls.print_patches(results.get_patches(), '\nSummary information about patches:')
-        for pkg in type_pkgs:
-            type_pkg = results.get_build(pkg)
-            if type_pkg:
-                cls.print_rpms(type_pkg, pkg.capitalize())
-                cls.print_build_logs(type_pkg, pkg.capitalize())
+            cls.print_patches(results.get_patches())
 
-        cls.print_pkgdiff_tool(results.get_checkers())
+        cls.print_message_and_separator("\nRPMS")
+        for pkg_version in ['old', 'new']:
+            pkg_results = results.get_build(pkg_version)
+            if pkg_results:
+                cls.print_rpms_and_logs(pkg_results, pkg_version.capitalize())
 
     @classmethod
-    def print_pkgdiff_tool(cls, checkers_results):
-        """Function prints a summary information about pkgcomparetool"""
+    def print_checkers_text_output(cls, checkers_results):
+        """Function prints text output for every checker"""
         if checkers_results:
-            for check, data in six.iteritems(checkers_results):
-                logger_report.info("=== Checker %s results ===", check)
-                if data:
-                    for checker, output in six.iteritems(data):
-                        if output is None:
-                            logger_report.info("Log is available here: %s\n", checker)
-                        else:
-                            if isinstance(output, list):
-                                logger_report.info("%s See for more details %s", ','.join(output), checker)
-                            else:
-                                logger_report.info("%s See for more details %s", output, checker)
+            for check_tool in six.itervalues(checkers_runner.plugin_classes):
+                for check, data in sorted(six.iteritems(checkers_results)):
+                    if check == check_tool.get_checker_name():
+                        logger_report.info(check_tool.format(data).rstrip())
```

Comment:
@nforro `rstrip` added here because of the extra line that logger adds anyway, which would cause inconsistency, as every line from checkers contains newline characters at the end and every headline from checker is preceeded with a newline character to logically divide the output sections.

Reply:
Well, maybe you could just return a list of lines (without additional whitespacing).

Reply:
Is this what you meant? 
https://github.com/rebase-helper/rebase-helper/pull/404/files#diff-3858a44859ef63796d6b6d43a35b9529R175

Reply:
Yes, but you should probably rename `output_string` variable to something more appropriate.